server/webhooks/vulnerabilities.go

```diff
@@ -57,41 +56,74 @@ func TriggerVulnerabilitiesWebhook(
 			if batchSize > 0 && len(hosts) > batchSize {
 				limit = batchSize
 			}
-			if err := sendVulnerabilityHostBatch(ctx, targetURL, v.CVE, serverURL, hosts[:limit], now); err != nil {
+
+			hostsPayload := getHostPayloadPart(serverURL, hosts[:limit])
+			payload := getVulnPayload(v, args.Meta[v.CVE], args.IsPremium, hostsPayload)
+
+			if err := sendVulnerabilityHostBatch(ctx, targetURL, payload, args.Time); err != nil {
 				return ctxerr.Wrap(ctx, err, "send vulnerability host batch")
 			}
+
 			hosts = hosts[limit:]
 		}
 	}
 
 	return nil
 }
 
-type vulnHostPayload struct {
-	ID       uint   `json:"id"`
-	Hostname string `json:"hostname"`
-	URL      string `json:"url"`
+type hostPayloadPart struct {
+	ID        uint   `json:"id"`
+	Hostname  string `json:"hostname"`
+	URL       string `json:"url"`
+	Platform  string `json:"platform"`
+	OSVersion string `json:"os_version"`
 }
 
-func sendVulnerabilityHostBatch(ctx context.Context, targetURL, cve string, hostBaseURL *url.URL, hosts []*fleet.HostShort, now time.Time) error {
-	shortHosts := make([]*vulnHostPayload, len(hosts))
+func getHostPayloadPart(hostBaseURL *url.URL, hosts []*fleet.HostShort) []*hostPayloadPart {
+	shortHosts := make([]*hostPayloadPart, len(hosts))
 	for i, h := range hosts {
 		hostURL := *hostBaseURL
 		hostURL.Path = path.Join(hostURL.Path, "hosts", strconv.Itoa(int(h.ID)))
-		shortHosts[i] = &vulnHostPayload{
-			ID:       h.ID,
-			Hostname: h.Hostname,
-			URL:      hostURL.String(),
+		shortHosts[i] = &hostPayloadPart{
+			ID:        h.ID,
+			Hostname:  h.Hostname,
+			URL:       hostURL.String(),
+			Platform:  h.Platform,
+			OSVersion: h.OSVersion,
 		}
 	}
+	return shortHosts
+}
+
+type payload struct {
+	CVE              string             `json:"cve"`
+	Link             string             `json:"details_link"`
+	EPSSProbability  *float64           `json:"epss_probability,omitempty"`
+	CVSSScore        *float64           `json:"cvss_score,omitempty"`
+	CISAKnownExploit *bool              `json:"cisa_known_exploit,omitempty"`
+	Hosts            []*hostPayloadPart `json:"hosts_affected"`
+}
+
+func getVulnPayload(vuln fleet.SoftwareVulnerability, meta fleet.CVEMeta, isPremium bool, hosts []*hostPayloadPart) payload {
+	r := payload{
+		CVE:   vuln.CVE,
+		Link:  fmt.Sprintf("https://nvd.nist.gov/vuln/detail/%s", vuln.CVE),
+		Hosts: hosts,
+	}
+
+	if isPremium {
+		r.EPSSProbability = meta.EPSSProbability
+		r.CVSSScore = meta.CVSSScore
+		r.CISAKnownExploit = meta.CISAKnownExploit
+	}
```

Comment:
the annoying thing is that we can't do this, all "premium" code must live under `ee/` for the license to work. Can this function (`getVulnPayload`) be a `service` function that gets enhanced in `ee/service`? You can take a look at the teams endpoint for examples, but the general idea is to do something like this in the premium service:

```go
func (svc *Service) getVulnPayload(...) payload {
    // retrieve common payload from the free service
    r := svc.Service.GetVulnPayload(...)
    
    // add premium fields
    r.EPSSProbability = meta.EPSSProbability
    r.CVSSScore = meta.CVSSScore
     r.CISAKnownExploit = meta.CISAKnownExploit
}
```

Reply:
Or I guess maybe `ee/server/webhoos`

Reply:
that would make sense to me, as long as we mimic what we do with service and "enchance" something with premium things if the license is premium